python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -592,39 +608,33 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None, fakeC
             # ExposureFitsReader
             fname = repoInfo.butler.getUri("calexp", dataRef.dataId)
             reader = afwImage.ExposureFitsReader(fname)
-            wcs = reader.readWcs()
+            detector = reader.readDetector()
+            if self.config.doApplyExternalSkyWcs:
+                wcs = dataRef.get(repoInfo.skyWcsDataset)
+            else:
+                wcs = reader.readWcs()
+            ccdCorners = wcs.pixelToSky(detector.getCorners(cameraGeom.PIXELS))
+            areaDict["corners_" + str(dataRef.dataId[repoInfo.ccdKey])] = ccdCorners
 
             # Actual mask is needed because BAD pixels are more than
             # just the defects
             mask = reader.readMask()
-
-            # Compute Focal Plane coordinates for each source if not already there
-            if self.config.hasFakes:
-
-                maskBad = mask.array & 2**mask.getMaskPlaneDict()['BAD']
-                good = np.count_nonzero(maskBad == 0)
-                pixScale = wcs.getPixelScale().asArcseconds()
-                area = pixScale**2 * good
-                areaDict[dataRef.dataId["ccd"]] = area
-
-            ccdWidth = mask.getWidth()
-            ccdHeight = mask.getHeight()
-
-            ccdCorners = wcs.pixelToSky([geom.Point2D(0, 0), geom.Point2D(ccdWidth, ccdHeight)])
-            areaDict["corners_" + str(dataRef.dataId[repoInfo.ccdKey])] = ccdCorners
+            maskBad = mask.array & 2**mask.getMaskPlaneDict()["BAD"]
+            maskNoData = mask.array & 2**mask.getMaskPlaneDict()["NO_DATA"]
+            maskedPixels = maskBad + maskNoData
+            numGoodPix = np.count_nonzero(maskedPixels == 0)
+            pixScale = wcs.getPixelScale().asArcseconds()
+            area = pixScale**2*numGoodPix
```

Comment:
I know it is against our conventions but is there a way to make pixScale**2*numGoodPix easier to parse at a glance? It's fine if there isn't I just find that the lack of white space makes it harder to understand at a glance, less relevant here but some more complicated maths bits suffer from this as well and I am curious.

Reply:
That solves my parsing problem, thank you!

Reply:
Please do not ever feel like you have to write something that everyone agrees is less readable to match conventions; [dev guide says so](https://developer.lsst.io/coding/intro.html#style-guide-deviations).  The spirit of the no-spaces-around-`*` rule is to indicate strong binding vs. `+` and `-`, and in the same spirit I think an expression involving `*` and `**` is an _excellent_ time to put spaces around `*`.